src/main/java/org/jabref/gui/exporter/ExportCommand.java

```diff
@@ -91,51 +92,21 @@ private void export(Path file, FileChooser.ExtensionFilter selectedExtensionFilt
         Globals.prefs.put(JabRefPreferences.EXPORT_WORKING_DIRECTORY, file.getParent().toString());
 
         final List<BibEntry> finEntries = entries;
-        AbstractWorker exportWorker = new AbstractWorker() {
-
-            String errorMessage;
-
-            @Override
-            public void run() {
-                try {
+        BackgroundTask
+                .wrap(() -> {
                     format.export(frame.getCurrentBasePanel().getBibDatabaseContext(),
                             file,
                             frame.getCurrentBasePanel()
-                                    .getBibDatabaseContext()
-                                    .getMetaData()
-                                    .getEncoding()
-                                    .orElse(Globals.prefs.getDefaultEncoding()),
+                                 .getBibDatabaseContext()
+                                 .getMetaData()
+                                 .getEncoding()
+                                 .orElse(Globals.prefs.getDefaultEncoding()),
                             finEntries);
-                } catch (Exception ex) {
-                    LOGGER.warn("Problem exporting", ex);
-                    if (ex.getMessage() == null) {
-                        errorMessage = ex.toString();
-                    } else {
-                        errorMessage = ex.getMessage();
-                    }
-                }
-            }
-
-            @Override
-            public void update() {
-                // No error message. Report success:
-                if (errorMessage == null) {
-                    frame.output(Localization.lang("%0 export successful", format.getDisplayName()));
-                }
-                // ... or show an error dialog:
-                else {
-                    frame.output(Localization.lang("Could not save file.") + " - " + errorMessage);
-                    // Need to warn the user that saving failed!
-                    frame.getDialogService().showErrorDialogAndWait(Localization.lang("Save library"), Localization.lang("Could not save file.") + "\n" + errorMessage);
-
-                }
-            }
-        };
-
-        // Run the export action in a background thread:
-        exportWorker.getWorker().run();
-        // Run the update method:
-        exportWorker.update();
+                    return null;
+                })
+                .onSuccess(x -> frame.output(Localization.lang("%0 export successful", format.getDisplayName())))
+                .onFailure(exception -> Platform.runLater(() -> handleError(exception)))
```

Comment:
this `runLater` shouldn't be necessary as `onSuccess` and `onFailure` are always run on the JavaFX thread (it would be nice if you could add this as a comment in BackgroundTask)